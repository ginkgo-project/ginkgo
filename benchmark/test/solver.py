#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]'],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.solver.json")],
    expected_stdout="solver.simple.stdout",
    expected_stderr="solver.simple.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt", "optimal": {"spmv": "csr"}}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="solver.profile.stdout",
    expected_stderr="solver.profile.stderr",
)
