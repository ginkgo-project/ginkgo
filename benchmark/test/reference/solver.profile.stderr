Running on reference(0)
Running with 0 warm iterations and 1 running iterations
The random seed for right hand sides is 42
Running cg with 1000 iterations and residual goal of 1.000000e-06
The number of right hand sides is 1
Running test case stencil(100, 7pt)
DEBUG: begin components::fill_array
DEBUG: end   components::fill_array
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin components::convert_idxs_to_ptrs
DEBUG: end   components::convert_idxs_to_ptrs
DEBUG: begin components::aos_to_soa
DEBUG: end   components::aos_to_soa
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill_in_matrix_data
DEBUG: end   dense::fill_in_matrix_data
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
Matrix is of size (125, 125)
DEBUG: begin stencil(100, 7pt)
	Running solver: cg
DEBUG: begin cg
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin copy
DEBUG: end   copy
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: begin repetition
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: begin generate(<typename>)
DEBUG: begin generate(<typename>)
DEBUG: end   generate(<typename>)
DEBUG: end   generate(<typename>)
DEBUG: begin apply(<typename>)
DEBUG: begin iteration
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin dense::fill
DEBUG: end   dense::fill
DEBUG: begin cg::initialize
DEBUG: end   cg::initialize
DEBUG: begin advanced_apply(<typename>)
DEBUG: begin csr::advanced_spmv
DEBUG: end   csr::advanced_spmv
DEBUG: end   advanced_apply(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: begin iteration
DEBUG: begin cg::step_1
DEBUG: end   cg::step_1
DEBUG: begin apply(<typename>)
DEBUG: begin csr::spmv
DEBUG: end   csr::spmv
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin cg::step_2
DEBUG: end   cg::step_2
DEBUG: begin apply(<typename>)
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: end   apply(<typename>)
DEBUG: begin dense::compute_conj_dot_dispatch
DEBUG: end   dense::compute_conj_dot_dispatch
DEBUG: begin check(<typename>)
DEBUG: begin check(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin residual_norm::residual_norm
DEBUG: end   residual_norm::residual_norm
DEBUG: end   check(<typename>)
DEBUG: end   check(<typename>)
DEBUG: end   iteration
DEBUG: end   apply(<typename>)
DEBUG: end   repetition
DEBUG: begin copy(<typename>)
DEBUG: begin dense::copy
DEBUG: end   dense::copy
DEBUG: end   copy(<typename>)
DEBUG: begin advanced_apply(<typename>)
DEBUG: begin csr::advanced_spmv
DEBUG: end   csr::advanced_spmv
DEBUG: end   advanced_apply(<typename>)
DEBUG: begin dense::compute_norm2_dispatch
DEBUG: end   dense::compute_norm2_dispatch
DEBUG: begin copy
DEBUG: end   copy
DEBUG: end   cg
DEBUG: end   stencil(100, 7pt)
